tests/coverage_test.ts

```diff
@@ -0,0 +1,114 @@
+
+    for (
+      const testcase of [
+        "anon",
+        "anon-async",
+        "class-method",
+        "named",
+        "named-async",
+        "misc_1",
+        "misc_2",
+      ] as const
+    ) {
+      await t.step(testcase, async () => {
+        // TODO(@lowlighter): to remove
+        const EXPECTED_DEBUG = {
+          anon: { "startOffset": 236, "endOffset": 392 },
+          "anon-async": { "startOffset": 236, "endOffset": 425 },
+          "class-method": { "startOffset": 284, "endOffset": 476 },
+          named: { "startOffset": 236, "endOffset": 403 },
+          "named-async": { "startOffset": 236, "endOffset": 436 },
+          misc_1: { "startOffset": 243, "endOffset": 406 },
+          misc_2: [{ "startOffset": 266, "endOffset": 440 }, {
+            "startOffset": 472,
+            "endOffset": 653,
+          }],
+        } as const;
+        console.log("expected offsets", EXPECTED_DEBUG[testcase]);
+        // ======================
+
+        await ensureDir(DENO_COVERAGE_DIR);
+        await emptyDir(DENO_COVERAGE_DIR);
+
+        const testFile = fromFileUrl(
+          import.meta.resolve(`./fixtures/coverage/${testcase}/test_main.ts`),
+        );
+        const scriptFile = fromFileUrl(
+          import.meta.resolve(`./fixtures/coverage/${testcase}/main.ts`),
+        );
+        const expectFile = fromFileUrl(
+          import.meta.resolve(`./fixtures/coverage/${testcase}/expect.out`),
+        );
+
+        // Run deno test --coverage
+        const denoTest = new Deno.Command(Deno.execPath(), {
+          args: [
+            "test",
+            "--allow-all",
+            `--coverage=${DENO_COVERAGE_DIR}`,
+            testFile,
+          ],
+          env: {
+            NO_COLOR: "true",
+          },
+        });
+        let output = await denoTest.output();
+        if (DEBUG) {
+          console.log(
+            decoder.decode(output.stdout).replace(
+              /ok \| 0 passed \| 0 failed[\s\S]+/,
+              "",
+            ),
+          );
+          console.log(decoder.decode(output.stderr));
+        }
+        assert(output.success, "deno test --coverage");
+
+        // Run deno coverage
+        const denoCoverage = new Deno.Command(Deno.execPath(), {
+          args: [
+            "coverage",
+            "--detailed",
+            `--include=${scriptFile}`,
+            DENO_COVERAGE_DIR,
+          ],
+          env: {
+            NO_COLOR: "true",
+          },
+        });
+        output = await denoCoverage.output();
+        assert(output.success, "deno coverage");
+
+        // Check that coverage files matches
+        const stdout = decoder.decode(output.stdout).trim();
+        if (DEBUG) {
+          console.log(stdout);
+        }
+        assertEquals(
+          stdout,
+          Deno.readTextFileSync(expectFile).replace(
+            "[import.meta.url]",
+            toFileUrl(scriptFile).href,
+          ).trim(),
+        );
```

Comment:
All files belows are fixtures to test the coverage feature so it's not necessary to perform in-depth review on them.
I've taken inspiration from the `specs` tests in denoland/deno, where you compare the expected deno output with what is generated in stdout

All the testcases are already correct. If all of them passes, then it's likely that the coverage feature is working as expected